### PR TITLE
feat: show license type in selection poc

### PR DESCRIPTION
**Pull request by marharyta**
Screenshot 2021-02-17 at 15 09 01

Requires to run ```lein reset``` and ```lein test-data``` to check with duplicate sources.

## Reviewability
- [x] Link to issue (#1999)
- [x] Consider adding screenshots for ease of review

## Backwards compatibility
- [ ] API is backwards compatible or completely new
- [x] Events are have migrations
- [ ] Config is backwards compatible

## Documentation
- [ ] Update changelog if necessary
- [ ] API is documented and shows up in Swagger UI
- [ ] Components are added to guide page
- [ ] Update docs/ (if applicable)
- [ ] Update manual/ (if applicable)
- [ ] ADR for major architectural decisions or experiments
- [ ] New config options in config-defaults.edn

## Testing
- [ ] Complex logic is unit tested
- [ ] Valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [ ] New tasks are created for pending or remaining tasks

### CODE REVIEW

File: dev-config.edn [hunk 1]
```diff
@@ -3,7 +3,7 @@
  :database-url "postgresql://localhost/rems?user=rems"
  :test-database-url "postgresql://localhost/rems_test?user=rems_test"
  :search-index-path "target/search-index-dev"
- :theme-path "example-theme/theme.edn"
+ :theme-path "../rems-deploy/themes/theme-thl/theme.edn"
```
Comment (opqdonut) [inline dev-config.edn:6]: you probably didn't mean to commit this

File: dev-config.edn [hunk 2]
```diff
@@ -16,7 +16,7 @@
  :oidc-extra-attributes [{:attribute "nickname" :name {:en "Nickname" :fi "Lempinimi"}}
                          {:attribute "picture"}]
 
- :languages [:en :fi :sv]
+ :languages [:en]
```
Comment (opqdonut) [inline dev-config.edn:19]: or this

File: src/clj/rems/db/test_data.clj [hunk 3]
```diff
@@ -859,10 +868,16 @@
 
         ;; Create resources
         res1 (create-resource! {:resource-ext-id "urn:nbn:fi:lb-201403262"
-                                :organization {:organization/id "nbn"}
+                                :organization {:organization/id "nbn"
+                                               :organization/short-name {:en "NBN en"
+                                                                         :fi "NBN fi"
+                                                                         :sv "NBN sv"}}
```
Comment (opqdonut) [inline src/clj/rems/db/test_data.clj:874]: `create-resource!` only cares about the `organization/id`. The other members of `:organization` are ignored. You can verify this by following the definitions to `rems.api.services.resource/create-resource!` if you want.

To explain that a bit, `create-resource!` only wants a reference to an existing organization, via a parameter like `:organization {:organization/id "foo"}`. It doesn't create the organization. The `:organization/short-name` is only needed when calling `create-organization!`.

File: src/clj/rems/db/test_data.clj [hunk 4]
```diff
@@ -859,10 +868,16 @@
...
                                 :actor owner})
         res2 (create-resource! {:resource-ext-id "Extra Data"
-                                :organization {:organization/id "nbn"}
+                                :organization {:organization/id "nbn"
+                                               :organization/short-name {:en "NBN en"
+                                                                         :fi "NBN fi"
+                                                                         :sv "NBN sv"}}
```
Comment (opqdonut) [inline src/clj/rems/db/test_data.clj:880]: same here

File: src/cljs/rems/administration/create_catalogue_item.cljs [hunk 5]
```diff
@@ -188,13 +188,34 @@
          :item-selected? item-selected?
          :on-change #(rf/dispatch [::set-selected-workflow %])}])]))
 
+(defn resourse-dropdown [r language counts]
+  (let [organisation? (get-in r [:organization :organization/short-name (keyword language)])
```
Comment (opqdonut) [inline src/cljs/rems/administration/create_catalogue_item.cljs:192]: I'm pretty sure the language is guaranteed to be a keyword, I think you can just replace `(keyword language)` with `language`.

File: src/cljs/rems/administration/create_catalogue_item.cljs [hunk 6]
```diff
@@ -188,13 +188,34 @@
          :item-selected? item-selected?
          :on-change #(rf/dispatch [::set-selected-workflow %])}])]))
 
+(defn resourse-dropdown [r language counts]
```
Comment (opqdonut) [inline src/cljs/rems/administration/create_catalogue_item.cljs:191]: Since this is used as an `:item-label`, perhaps a name like `resource-label`? Also, it could be private.

File: src/cljs/rems/administration/create_catalogue_item.cljs [hunk 7]
```diff
@@ -188,13 +188,34 @@
...
+         (when organisation?
+           (str/join " " [" (org:" organisation? ")"]))
+      ;; when duplicate is found, show duplicate
+         (when duplicate?
+         ;; when licenses are found, show licences, but only if it is duplicate
+           (when licenses?
+             (str/join " "
+                       [" (licenses:"
+                        (str/join ", " (mapv
+                                        (fn [l] (:title ((keyword language) (:localizations l))))
```
Comment (opqdonut) [inline src/cljs/rems/administration/create_catalogue_item.cljs:206]: consider: `(fn [l] (get-in l [:localizations language :title]))`

File: src/cljs/rems/administration/create_catalogue_item.cljs [hunk 8]
```diff
@@ -188,13 +188,34 @@
...
+
 (defn- catalogue-item-resource-field []
   (let [resources @(rf/subscribe [::resources])
+        counts (frequencies (map :resid resources))
         editing? @(rf/subscribe [::editing?])
         selected-resource @(rf/subscribe [::selected-resource])
         item-selected? #(= (:id %) (:id selected-resource))
         language @(rf/subscribe [:language])]
     [:div.form-group
+     (js/console.log "yay" (clj->js (get-in (first resources) [:organization :id])))
```
Comment (opqdonut) [inline src/cljs/rems/administration/create_catalogue_item.cljs:218]: this can probably be left out

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 9]
```diff
@@ -46,3 +46,82 @@
...
+                                :enabled true
+                                :id 8
+                                :licensetype "text"
+                                :localizations {:en {:attachment-id nil
+                                                     :textcontent "Be fast."
+                                                     :title "Performance License"}}}]}
+                   {:resid "u"}
+                   {:resid "z"}]
+        counts (frequencies (map :resid resourses))]
+    (testing "sorting resources"
```
Comment (opqdonut) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:112]: is this test about sorting?

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 10]
```diff
@@ -46,3 +46,82 @@
...
+      (is (not (empty? (map resourse-dropdown resourses))))
```
Comment (opqdonut) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:113]: this test case looks a bit weird: it's calling `resourse-dropdown` with only one parameter, when it really wants 3, right?

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 11]
```diff
@@ -46,3 +46,82 @@
...
+      (is (= ["x (org: NBN en )"
+              "y (org: NBN en ) (licenses: Performance License, Second License )"
```
Comment (opqdonut) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:115]: Could you get rid of the extra space? I mean `(org: NBN en)` instead of `(org: NBN en )` and similarly for `(licenses: ...)`. Just something that bugs me.

**Comment by marharyta** [comment]
@opqdonut I could potentially add more tests, but this far I think this is the minimum that does the job

**Review by opqdonut — CHANGES_REQUESTED** [review]
In general, a nice surgical implementation, and good tests. Thanks. Some stuff to clean up however, see my comments.

**Comment by opqdonut** [comment]
> @opqdonut I could potentially add more tests, but this far I think this is the minimum that does the job

I think the current test is enough!

File: dev-config.edn (hunk 1 above)
Comment (marharyta) [reply dev-config.edn:6]: oh, true, my bad

File: dev-config.edn (hunk 2 above)
Comment (marharyta) [reply dev-config.edn:19]: same here

File: src/clj/rems/db/test_data.clj (hunk 3 above)
Comment (marharyta) [reply src/clj/rems/db/test_data.clj:874]: I ll fix this, I am pretty sure you mentioned it last time but it somehow escaped my memory

File: src/cljs/rems/administration/create_catalogue_item.cljs (hunk 7 above)
Comment (marharyta) [reply src/cljs/rems/administration/create_catalogue_item.cljs:206]: That could be a better way to do it, thanks!

File: src/cljs/rems/administration/create_catalogue_item.cljs (hunk 8 above)
Comment (marharyta) [reply src/cljs/rems/administration/create_catalogue_item.cljs:218]: yes, my bad

File: test/cljs/rems/administration/test_create_catalogue_item.cljs (hunk 11 above)
Comment (marharyta) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:115]: sure, it was bugging me too, but I wanted to very this with you

File: src/cljs/rems/administration/create_catalogue_item.cljs [hunk 12]
```diff
@@ -188,13 +188,34 @@
          :item-selected? item-selected?
          :on-change #(rf/dispatch [::set-selected-workflow %])}])]))
 
+(defn resourse-dropdown [r language counts]
+  (let [organisation? (get-in r [:organization :organization/short-name (keyword language)])
+        duplicate? (> (get counts (:resid r)) 1)
+        licenses? (not-empty (:licenses r))]
+    (str (:resid r)
+      ;; when organisation is found, show organisation
```
Comment (opqdonut) [inline src/cljs/rems/administration/create_catalogue_item.cljs:196]: Oh, one more thing: I think your code is clear enough even without these comments. What do you think?

File: src/cljs/rems/administration/create_catalogue_item.cljs [hunk 13]
```diff
@@ -188,8 +188,24 @@
          :item-selected? item-selected?
          :on-change #(rf/dispatch [::set-selected-workflow %])}])]))
 
+(defn resource-label [r language counts]
+  (let [organisation (get-in r [:organization :organization/short-name language])
+        duplicate? (> (get counts (:resid r)) 1)
+        licenses? (not-empty (:licenses r))]
+    (str (:resid r)
+         (when organisation
+           (str/join [" (" (text :t.administration/org) ": " organisation ")"]))
```
Comment (opqdonut) [inline src/cljs/rems/administration/create_catalogue_item.cljs:197]: `(str/join [x y z])` can be written more simply as `(str x y z)`

File: src/cljs/rems/administration/create_catalogue_item.cljs [hunk 14]
```diff
@@ -188,8 +188,24 @@
...
+         (when duplicate?
+           (when licenses?
+             (str/join [" (" (str/lower-case (text :t.administration/licenses)) ": "
```
Comment (opqdonut) [inline src/cljs/rems/administration/create_catalogue_item.cljs:200]: this `str/join` can be simplified as well

File: src/cljs/rems/administration/create_catalogue_item.cljs [hunk 15]
```diff
@@ -188,8 +188,24 @@
...
+                        (str/join ", " (mapv
```
Comment (opqdonut) [inline src/cljs/rems/administration/create_catalogue_item.cljs:201]: this one can't since it uses a delimiter

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 16]
```diff
@@ -46,3 +46,82 @@
     (testing "missing form"
       (is (nil? (build-request (assoc form :form nil)
                                languages))))))
+
+(deftest resourse-dropdown-test
```
Comment (opqdonut) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:50]: nit: this should probably be `resource-label-test` to match the name of the function

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 17]
```diff
@@ -46,3 +46,82 @@
...
+  (let [resourses [{:resid "x"
```
Comment (opqdonut) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:51]: nit: resource has a C

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 18]
```diff
@@ -46,3 +46,82 @@
...
+                                :id 8
+                                :licensetype "text"
+                                :localizations {:en {:attachment-id nil
+                                                     :textcontent "Be fast."
+                                                     :title "Performance License"}}}]}
+                   {:resid "u"}
+                   {:resid "z"}]
+        counts (frequencies (map :resid resourses))]
+    (testing "resource-label"
+      (is (not (empty? (map #(resource-label % :en counts) resourses))))
```
Comment (opqdonut) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:113]: it seems to me this test is redundant with the following one: both run the same code, but this one only checks that the result is non-empty, while the next one checks the content literally

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 19]
```diff
@@ -46,3 +46,82 @@
...
+      (is (= ["x ([:t.administration/org]: NBN en)"
+              "y ([:t.administration/org]: NBN en) ([:t.administration/licenses]: Performance License, Second License)"
+              "y ([:t.administration/licenses]: Performance License)"
+              "u"
+              "z"]
+             (map #(resource-label % :en counts) resourses)))
+      (is (not (= ["x (org: NBN fi)"
```
Comment (opqdonut) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:120]: are you sure you want `(not (=` here?

**Review by opqdonut — CHANGES_REQUESTED** [review]
looks good, some minor points to fix

File: src/cljs/rems/administration/create_catalogue_item.cljs (hunk 13 above)
Comment (marharyta) [reply src/cljs/rems/administration/create_catalogue_item.cljs:197]: oh, I didn't know that! Cool!

File: test/cljs/rems/administration/test_create_catalogue_item.cljs (hunk 16 above)
Comment (marharyta) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:50]: yeah, right, I missed it in the refactoring

File: src/cljs/rems/administration/create_catalogue_item.cljs (hunk 13 above)
Comment (marharyta) [reply src/cljs/rems/administration/create_catalogue_item.cljs:197]: I will not be refactoring this since it creates additional friction with testing, but overall it is a nice suggestion

Comment (marharyta) [reply src/cljs/rems/administration/create_catalogue_item.cljs:202]: - replace nested whens with and
- remove string joins and replace with just one statement

File: src/cljs/rems/administration/create_catalogue_item.cljs [hunk 20]
```diff
@@ -188,8 +188,30 @@
...
+                     false)]
+    (str (:resid r)
+         (when organisation
+           (str/join [" (" (text :t.administration/org) ": " organisation ")"]))
+         (when duplicate?
+           (when licenses?
+             (when localized?
+               (str/join [" (" (str/lower-case (text :t.administration/licenses)) ": "
+                          (str/join ", " (filter #(not (nil? %)) (mapv
+                                                                  (fn [l] (get-in l [:localizations language :title]))
```
Comment (marharyta) [inline src/cljs/rems/administration/create_catalogue_item.cljs:208]: - remove instead of filter not nil

Comment (marharyta) [reply src/cljs/rems/administration/create_catalogue_item.cljs:208]: - license translation
- capitalize org

**Comment by marharyta** [comment]
@Macroz re-request ping

File: src/cljs/rems/administration/create_catalogue_item.cljs [hunk 21]
```diff
@@ -188,8 +188,22 @@
          :item-selected? item-selected?
          :on-change #(rf/dispatch [::set-selected-workflow %])}])]))
 
+(defn resource-label [r language counts]
+  (let [organisation (get-in r [:organization :organization/short-name language])
+        duplicate? (> (get counts (:resid r)) 1)
+        licenses? (not-empty (:licenses r))]
```
Comment (Macroz) [inline src/cljs/rems/administration/create_catalogue_item.cljs:194]: The Clojure idiom for this is `(seq (:licenses r))`

Comment (marharyta) [reply src/cljs/rems/administration/create_catalogue_item.cljs:194]: Thanks!

**Comment by marharyta** [comment]
@Macroz I believe that should do it?

**Comment by marharyta** [comment]
Screenshot 2021-02-24 at 16 40 29

File: src/cljs/rems/administration/create_catalogue_item.cljs [hunk 22]
```diff
@@ -204,10 +218,7 @@
         {:id resource-dropdown-id
          :items (->> resources (filter :enabled) (remove :archived))
          :item-key :id
-         :item-label #(str (:resid %)
-                           " (org: "
-                           (get-in % [:organization :organization/short-name language])
-                           ")")
+         :item-label #(resource-label % language counts)
```
Comment (Macroz) [inline src/cljs/rems/administration/create_catalogue_item.cljs:221]: Minor thing here, I would prefer to send the count of the one resource only here, not the counts of all resources to the `resource-label` function. The responsibilities are a bit mixed. I'm not insisting on changing this.

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 23]
```diff
@@ -46,3 +46,105 @@
...
+                                                     :title "Performance License"}
+                                                :fi {:attachment-id nil
+                                                     :textcontent "Be fast."
+                                                     :title "Performance License FI"}
+                                                :sv {:attachment-id nil
+                                                     :textcontent "Be fast."
+                                                     :title "Performance License SV"}}}]}
+                   {:resid "u"}
+                   {:resid "z"}]
+        counts (frequencies (map :resid resources))]
```
Comment (Macroz) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:129]: And here, you would not have to do this if you sent the single count. This duplicates "complex logic".

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 24]
```diff
@@ -46,3 +46,105 @@
     (testing "missing form"
       (is (nil? (build-request (assoc form :form nil)
                                languages))))))
+
+(deftest resource-label-test
```
Comment (Macroz) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:50]: I think this test has some problems that we could work on:
- First of all, this test is massive. We should not need to add almost 100 lines of example data to test this tiny feature. How about we reduce the resource test data to only what one call to `resource-label` actually needs? Second possibility is to refactor or extract the complex logic in another manner so that testing it becomes easier. Another possibility is writing this level of test as an integration test with real created resources. I think we should do at least the first one.
- The expected expressions contain a mix of language and unlocalized keys. I think it smells of wrong level of testing or functions. I think we can live with this for now.
- Why does the test of `resource-label` function need to use `map` and call the function to several resources? Would it not be better to e.g. test one resource at a time each in its own `is` expression with possible explanation as to what case it is?
- Why does it need to test all the different languages and why is not one enough?

**Review by Macroz — DISMISSED** [review]
I think this looks pretty good now. The changes you did to the logic of `resource-label` worked out. I think we should spend a little love on the test now. It is massive! If you try the first suggestion (i.e. removing unnecessary parts of the test data and simplifying the names of things) and then streamline the actual test assertions I think it'll be good enough.

File: test/cljs/rems/administration/test_create_catalogue_item.cljs (hunk 24 above)
Comment (marharyta) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:50]: I agree with you that the test could use some clean-up. The pointer is pretty clear. I ll work on the test data a bit more and come back to you.

Comment (marharyta) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:50]: "Why does it need to test all the different languages and why is not one enough? " - in the previous iterations I was not sure whether all license features were localized so adding that made sense

Comment (marharyta) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:50]: "First of all, this test is massive. We should not need to add almost 100 lines of example data to test this tiny feature. How about we reduce the resource test data to only what one call to resource-label actually needs? " - done

File: src/cljs/rems/administration/create_catalogue_item.cljs (hunk 22 above)
Comment (marharyta) [reply src/cljs/rems/administration/create_catalogue_item.cljs:221]: I am not sure what implications it might have, maybe I can leave it like this for now?

File: test/cljs/rems/administration/test_create_catalogue_item.cljs (hunk 23 above)
Comment (marharyta) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:77]: I do not understand this comment, I am sorry

File: test/cljs/rems/administration/test_create_catalogue_item.cljs (hunk 24 above)
Comment (marharyta) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:50]: Why does the test of resource-label function need to use map and call the function to several resources? - isn't the entire purpose of the function to iterate through the selections?

File: test/cljs/rems/administration/test_create_catalogue_item.cljs (hunk 23 above)
Comment (Macroz) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:77]: Just that if you did the previous, and if you could pass the count simply to `resource-label` as a number, you would not need to duplicate this code in the test, that you must do in the real component.

File: test/cljs/rems/administration/test_create_catalogue_item.cljs (hunk 24 above)
Comment (Macroz) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:50]: First of all you are testing `resource-label` i.e. the label for a *single resource*. Therefore the test should test that. And that does not iterate through selections, right?

**Comment by marharyta** [comment]
@Macroz I think I have addressed everything at this stage?

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 25]
```diff
@@ -76,6 +76,34 @@
                    {:resid "z"}]
         counts (frequencies (map :resid resources))]
     (testing "resource-label"
+      (is (= ["y ([:t.administration/org]: NBN en) ([:t.administration/licenses]: Performance License, Second License)"
+              "y ([:t.administration/licenses]: Performance License)"]
+             (map #(resource-label % :en counts) [{:resid "y"
```
Comment (Macroz) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:81]: Why still the `map` here? Shouldn't the second be its own case like the second `is` below?

Comment (marharyta) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:81]: I wanted to check the case with two similar resids. 
I can add another case below.

Comment (Macroz) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:81]: One `is` per one call of tested function i.e. `resource-label` is the normal case. Also now you already moved half a step into this direction but not quite there.

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 26]
```diff
@@ -104,6 +104,19 @@
...
+                              :licenses [{:localizations {:en {:title "Performance License"}
+                                                          :fi {:title "Performance License FI"}
+                                                          :sv {:title "Performance License SV"}}}
+                                         {:localizations {:en {:title "Second License"}
+                                                          :fi {:title "Second License FI"}
+                                                          :sv {:title "Second License SV"}}}]
+                              :organization {:organization/id "nbn"
+                                             :organization/short-name {:en "NBN en"
+                                                                       :fi "NBN fi"
+                                                                       :sv "NBN sv"}}} :en counts)))
```
Comment (opqdonut) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:118]: I like an explicit test like this. you could also have a literal `{"y" 2}` for `counts`. Then you could only have literal cases like this, and wouldn't need the `let` at the start of this test at all.

File: test/cljs/rems/administration/test_create_catalogue_item.cljs [hunk 27]
```diff
@@ -46,3 +46,107 @@
...
+                    :licenses [{:localizations {:en {:title "Performance License"}
+                                                :fi {:title "Performance License FI"}
+                                                :sv {:title "Performance License SV"}}}]}
+                   {:resid "u"}
+                   {:resid "z"}]
+        counts (frequencies (map :resid resources))]
+    (testing "resource-label"
+      (is (= ["y ([:t.administration/org]: NBN en) ([:t.administration/licenses]: Performance License, Second License)"
+              "y ([:t.administration/licenses]: Performance License)"]
+             (map #(resource-label % :en counts) [{:resid "y"
```
Comment (Macroz) [inline test/cljs/rems/administration/test_create_catalogue_item.cljs:81]: And the `map` is still here?

Comment (marharyta) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:81]: I did not delete it, I added more tests cases.

**Comment by marharyta** [comment]
ping @Macroz @opqdonut re-review

File: test/cljs/rems/administration/test_create_catalogue_item.cljs (hunk 27 above)
Comment (Macroz) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:81]: You marked the discussion as resolved?

Comment (marharyta) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs:81]: this one? no, obviously

Comment (marharyta) [reply test/cljs/rems/administration/test_create_catalogue_item.cljs]: I can delete the test altogether

**Review by opqdonut — APPROVED** [review]

**Comment by marharyta** [comment]
btw, @opqdonut  and @Macroz  if there is something that prevents the PR from being merged, please be a bit more explicit about it. If the fix is preventing the PR from being merged, it is not a suggestion, it is a requirement. I am completely ok with them, but please put that distinction out there so that it is clear to me.
